### DISCUSSION TITLE
Adding a link to the upper layer in AnimLayer?

### DISCUSSION THREAD

**Discussion by Ali-RS** [jMonkeyEngine/jmonkeyengine#2048] [Ideas]
Hello,

Is it a good idea to add a link to the upper layer in the [AnimLayer](https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-core/src/main/java/com/jme3/anim/AnimLayer.java) class? So by accessing an AnimLayer, we can walk through all its above layers.

Linking should be done automatically by AnimComposer whenever a new layer is added or an existing layer is removed.

Required methods:

`AnimLayer.setUpperLayer()/getUpperLayer()`

What is the use case?

An example use case would be in this PR https://github.com/jMonkeyEngine/jmonkeyengine/pull/2043 where we need to find all the above layers from a specific layer.

something like

```
while((layer = layer.getUpperLayer()) != null) {
   // do stuff
}
```

what do you think?

**Comment by pavly-gerges** [comment]
Hi, You guys are opening discussions here, nice, I did see it by chance :-).

> Is it a good idea to add a link to the upper layer in the [AnimLayer](https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-core/src/main/java/com/jme3/anim/AnimLayer.java) class? So by accessing an AnimLayer, we can walk through all its above layers.

That's a type of Visitor Pattern, it's very useful, though, I think it already exists through the AnimComposer object, if that is true, then no need to replicate new encapsulations, just provide a use-case.
